src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/QueueListenerImpl.java

```diff
@@ -58,6 +65,33 @@ public void onEnterWaiting(Queue.WaitingItem wi) {
         EiffelActivityTriggeredEvent.Data data = new EiffelActivityTriggeredEvent.Data(taskName);
         EiffelActivityTriggeredEvent event = new EiffelActivityTriggeredEvent(data);
         EiffelJobTable.getInstance().setEventTrigger(wi.getId(), event.getMeta().getId());
+
+        for (Cause cause : wi.getCauses()) {
+            // Default to OTHER as the default trigger type and override it for specific known causes.
+            EiffelActivityTriggeredEvent.Data.Trigger trigger = new EiffelActivityTriggeredEvent.Data.Trigger(
+                    EiffelActivityTriggeredEvent.Data.Trigger.Type.OTHER);
+            trigger.setDescription(cause.getShortDescription());
+
+            if (cause instanceof TimerTrigger.TimerTriggerCause) {
+                trigger.setType(EiffelActivityTriggeredEvent.Data.Trigger.Type.TIMER);
+            } else if (cause instanceof SCMTrigger.SCMTriggerCause) {
+                trigger.setType(EiffelActivityTriggeredEvent.Data.Trigger.Type.SOURCE_CHANGE);
+            } else if (cause instanceof Cause.UpstreamCause) {
+                UUID triggerId = getUpstreamTriggerId((Cause.UpstreamCause) cause);
+                if (triggerId != null) {
+                    event.getLinks().add(new EiffelEvent.Link(EiffelEvent.Link.Type.CAUSE, triggerId));
+                    trigger.setType(EiffelActivityTriggeredEvent.Data.Trigger.Type.EIFFEL_EVENT);
+                }
```

Comment:
Would be nice if we could break these "case instanceof" into a static method, so we can have some reasonable tests about expected outputs.

Reply:
Edit: I see you went with a nice hasTrigger() matcher instead. That's fine.

Reply:
There are no tests for the timer and SCM cause cases though. It would be easy to add but it's pretty trivial functionality we'd be testing so I ignored it and figured that if UserIdCause and UpstreamCause work as expected the rest are probably fine.

Reply:
Ack